src/snowcli/cli/project/config.py

```diff
@@ -0,0 +1,76 @@
+from .util import clean_identifier, get_env_username
+from pathlib import Path
+from typing import List
+from strictyaml import (
+    YAML,
+    load,
+    as_document,
+)
+
+from .schema import (
+    project_schema,
+    project_override_schema,
+)
+
+from snowcli.cli.common.snow_cli_global_context import snow_cli_global_context_manager
+
+DEFAULT_USERNAME = "unknown_user"
+
+
+def merge_left(target: dict | YAML, source: dict | YAML) -> None:
+    """
+    Recursively merges key/value pairs from source into target.
+    Modifies the original dict-like "target".
+    """
+    for k, v in source.items():
+        if k in target and (isinstance(v, dict) or isinstance(v, YAML)):
+            # assumption: all inputs have been validated.
+            assert isinstance(target[k], dict) or isinstance(target[k], YAML)
+            merge_left(target[k], v)
+        else:
+            target[k] = v
+
+
+def load_project_config(paths: List[Path]) -> dict:
+    """
+    Loads a project config, optionally overriding values. Configuration is merged
+    in order of left to right (increasing precedence).
+    """
+    if len(paths) == 0:
+        raise ValueError("Need at least one configuration file.")
+
+    with open(paths[0], "r") as base_yml:
+        config = load(base_yml.read(), project_schema)
+
+    for override_path in paths[1:]:
+        with open(override_path, "r") as override_yml:
+            overrides = load(override_yml.read(), project_override_schema)
+            merge_left(config, overrides)
+
+        # TODO: how to show good error messages here?
+        config.revalidate(project_schema)
+
+    return config.data
+
+
+def generate_local_override_yml(project: dict | YAML) -> YAML:
+    user = clean_identifier(get_env_username() or DEFAULT_USERNAME)
+    role = snow_cli_global_context_manager.get_single_value("select current_role()")
+    warehouse = snow_cli_global_context_manager.get_single_value(
+        "select current_warehouse()"
+    )
```

Comment:
This can be probably extracted to more general functionality.  For example it can be useful when adding new connection to populate defaults fro warehouse and role.  WDYT?

Reply:
Could you expand on that use case? Are you thinking about building a new connection profile based on an existing live connection here?